runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7715,6 +7715,21 @@ class INTERPRETER_CLASS
 		/* Swap flags and class subfield order. */
 		classAndFlags = J9CLASSANDFLAGS_FROM_FLAGSANDCLASS(flagsAndClass);
 		valueAddress = J9STATICADDRESS(flagsAndClass, valueOffset);
+
+#if defined(J9VM_OPT_VALHALLA_STRICT_FIELDS)
+		if (J9ClassInitNotInitialized == (ramConstantPool->ramClass->initializeStatus & J9ClassInitStatusMask)) {
+			if (J9_STATIC_FIELD_STRICT_INIT_IS_UNSET(classAndFlags)) {
+				/* If a strict field has never been set, fail. */
+				rc = THROW_RUNTIME_EXCEPTION;
+				goto done;
+			} else if (J9_STATIC_FIELD_STRICT_INIT_WAS_WRITTEN(classAndFlags)) {
+				classAndFlags &= ~J9StaticFieldRefStrictInitUnset;
```

Comment:
All places where the strict-init subfield is modified should use `J9StaticFieldRefStrictInitMask`:
```c
				classAndFlags &= ~J9StaticFieldRefStrictInitMask;
				classAndFlags |= J9StaticFieldRefStrictInitRead;
```
The removes the assumption that `J9StaticFieldRefStrictInitUnset` has all bits set.